### DISCUSSION TITLE
backend.ai-manager의 Docker registry setup TimeOut Error

### DISCUSSION THREAD

**Discussion by TeusEE** [lablup/contribution-academy-2021#70] [Q&A]
**Describe the problem**
install-dev.sh의 643~650 line의 t실행 과정에서
docker images registry를 가지고오는 Process가 6분이 초과되면 TimeOut Error가 발생
해당 문제 때문에 client 모드에서 backend.ai admin images를 실행하면 아무런 이미지가 없다고 나오며,
정상적인 테스트 불가능
Stable, Commnuity 모두 설정했을 때와 Stable만 설정했을 때 오류가 나는 위치가 다른걸로 봐선 TimeOut 문제로 추정되어 보입니다.

**To Reproduce**
install-dev.sh 실행 -> 643~650 Line 실행 중 Error 발생 후 해당 부분 Skip한 뒤 설치 완료
-> cd "${INSTALL_PATH}/manager"
python -m ai.backend.manager.cli etcd put config/docker/registry/cr.backend.ai "https://cr.backend.ai"
python -m ai.backend.manager.cli etcd put config/docker/registry/cr.backend.ai/type "harbor2"
python -m ai.backend.manager.cli etcd put config/docker/registry/cr.backend.ai/project "stable,community"
python -m ai.backend.manager.cli etcd rescan-images cr.backend.ai

**Screenshots**
![image](https://user-images.githubusercontent.com/79349894/130232645-a5e403f6-1a11-4361-b682-96363c820be4.png)

**Environment Description**
OS: Windows10 WSL2 Ubuntu20.04
Current Shell: bash

**Comment by lizable** [comment]
@TeusEE 
I need to inspect this more. While then, Could you follow these steps and attach a screenshot or responses you got?
1. change directory to `manager`.
`cd ../manager` (../ is not mandatory, can be changed via current directory of your local environment)
2. Execute command below:
   > ⚠️ NOTICE: make sure the pyenv is enabled, if not backend.ai related command will throw "command not found error".

   `backend.ai mgr etcd get --prefix config/docker`

**Reply by TeusEE** [reply]
manager로 이동하여 pyenv 가상환경 내에서 실행한 결과이며
docker config 확인하였을때 문제는 없는것으로 확인됩니다.
전반적으로 6분 초과에서 짤리는것을 보아, TimeOut문제가 유력해보이는거같습니다

1. Stable과 community 둘다 포함되었을 경우
![image](https://user-images.githubusercontent.com/79349894/130307762-2afeb856-d7cd-499e-bca8-2bfd1e77cf50.png)


2. Stable만 진행했을 경우
![image](https://user-images.githubusercontent.com/79349894/130307816-174e1ce4-462a-4d20-a724-affe8bd8ca81.png)


3. community만 진행했을 경우
![image](https://user-images.githubusercontent.com/79349894/130308031-1ea6eabb-ab4d-444c-9667-0c3bcd52e719.png)
